src/examples/millionairs_problem/common/millionair.cpp

```diff
@@ -0,0 +1,83 @@
+// MIT License
+//
+// Copyright (c) 2019 Oleksandr Tkachenko, Daniel Demmler, Tobias Kussel
+// Cryptography and Privacy Engineering Group (ENCRYPTO)
+// TU Darmstadt, Germany
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "millionair.h"
+
+#include <bits/c++config.h>
+#include <cstddef>
+#include <vector>
+#include <limits>
+
+#include "algorithm/algorithm_description.h"
+#include "protocols/boolean_gmw/boolean_gmw_wire.h"
+#include "protocols/share_wrapper.h"
+#include "secure_type/secure_unsigned_integer.h"
+#include "utility/config.h"
+
+// abbreviate namespace
+namespace mo = encrypto::motion;
+
+void EvaluateProtocol(mo::PartyPointer& party, std::uint32_t value) {
+  const std::size_t number_of_parties{party->GetConfiguration()->GetNumOfParties()};
+
+  // (pre-)allocate indices and input values
+  std::vector<mo::SecureUnsignedInteger> indices(number_of_parties), input_values(number_of_parties);
+  
+  // share inputs
+  // remark: the input values to other parties' input gates are considered as buffers 
+  // and the values are simply ignored and overwritten later
```

Comment:
@Oleksandr-Tkachenko Is there still no better interface on master? (not sure about this)

Reply:
1) There is no better interface on master yet.
2) We could add this interface, should not be a problem.
3) I like this option in this case more because we don't need this:
```
if(id == my_id){
  this_input = MyInput(my_real_input);
} else {
  this_input = DummyInput(input_owner_id, input_size);
}
```